go/fn/doc.go

```diff
@@ -13,58 +13,66 @@
 // limitations under the License.
 
 /*
-Package fn.provides an SDK for writing KRM functions in Go. The function
-specification is defined at:
-https://github.com/kubernetes-sigs/kustomize/blob/master/cmd/config/docs/api-conventions/functions-spec.md
+Package fn provides the SDK to write KRM functions.
 
-A KRM functions can generate, mutate or validate Kubernetes resources in a
-ResourceList.
+Before you start
 
-The ResourceList type and the KubeObject type are the core part of this package.
+This fn SDK requires some basic KRM function Specification knowledge. To make the best usage of your time, we recommend
+you to be familiar with "ResourceList" before moving forward.
+
+  The KRM Function Specification, or "ResourceList", defines the standards of the inter-process communication between
+  the orchestrator (i.e. kpt CLI) and functions.
+
+It is created by kpt project and later on donated to CNCF.
```

Comment:
I'm not sure if this is true.
IIRC it was created in kustomize repo and only kpt adopt it at the time.
This sentence is not very relevant, IMO we can drop it.

Reply:
oh that's from the kpt book by Frank. https://kpt.dev/book/05-developing-functions/01-functions-specification

Sure. If it causes doubt, I can drop it. But I think at that time function was called "kpt function". It has a reason. 😄